django/db/models/fields/related_descriptors.py

```diff
@@ -896,6 +896,7 @@ def get_queryset(self):
                 return self.instance._prefetched_objects_cache[self.prefetch_cache_name]
             except (AttributeError, KeyError):
                 queryset = super().get_queryset()
+                queryset._deferred_filters_enabled = True
```

Comment:
Something similar should be done in `prefetch_one_level` before calling `_apply_rel_filters` when a `Prefetch(queryset)` is provided. Of maybe it should be done in `_apply_rel_filters` itself.

Reply:
Chose the latter option. Thanks.